### ISSUE TITLE
[Bug]: UX issues when requesting share by mail password

### ISSUE THREAD

**Issue by PVince81** [nextcloud/server#31952]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

At step 5 the user is more likely to quickly click the "Request password" button instead of the little arrow to submit the current one.

After step 7, an extra click is needed to go back.

### Steps to reproduce

1. Share a folder by mail
2. Set a password for that share
3. Log out
4. Open the link from the email
5. Quickly enter the known password and click the submit button: here I often click the **big** button instead of the arrow out of reflex
6. Click on "Request password"
7. Enter email address and submit
8. Be frustrated to have to click "Back"

### Expected behavior

At step 5, find a way to make the "Request password" button less visible. Or maybe allow the user to first enter a wrong password and only then show the button ? @jancborchardt thoughts ?

At step 7, as soon as the email is set, redirect back to the password field and show a message there "The password was sent by email". This way the user can copy paste and directly enter instead of having to go back.

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### List of activated Apps

```shell
Enabled:
  - accessibility: 1.10.0
  - activity: 2.16.0
  - cloud_federation_api: 1.7.0
  - comments: 1.14.0
  - contactsinteraction: 1.5.0
  - dashboard: 7.4.0
  - dav: 1.22.0
  - federatedfilesharing: 1.14.0
  - federation: 1.14.0
  - files: 1.19.0
  - files_external: 1.16.1
  - files_sharing: 1.16.2
  - files_trashbin: 1.14.0
  - files_versions: 1.17.0
  - lookup_server_connector: 1.12.0
  - nextcloud_announcements: 1.13.0
  - oauth2: 1.12.0
  - photos: 1.6.0
  - provisioning_api: 1.14.0
  - recommendations: 1.3.0
  - serverinfo: 1.14.0
  - settings: 1.6.0
  - sharebymail: 1.14.0
  - survey_client: 1.12.0
  - systemtags: 1.14.0
  - theming: 1.15.0
  - twofactor_backupcodes: 1.13.0
  - updatenotification: 1.14.0
  - user_status: 1.4.0
  - viewer: 1.8.0
  - weather_status: 1.4.0
  - workflowengine: 2.6.0
Disabled:
  - admin_audit
  - bruteforcesettings
  - calendar
  - contacts
  - deck
  - email_template_example-0.0.1
  - encryption
  - files_3d
  - files_accesscontrol
  - files_antivirus
  - files_automatedtagging
  - files_downloadlimit
  - files_retention
  - files_texteditor
  - files_versions_s3
  - groupfolders
  - guests
  - mail
  - music
  - officeonline
  - password_policy: 1.14.0
  - previewgenerator
  - ransomware_protection
  - registration
  - richdocuments
  - shareimporter
  - sharepermissions
  - spreed
  - support: 1.7.0
  - suspicious_login
  - terms_of_service
  - testing
  - twofactor_totp
  - user_ldap
  - user_migration
  - user_saml
  - workflow_script
```

### Additional info

master is df14579173708b380f75a6f5680748e37fdbc419

**Comment by StCyr** [comment]
> At step 5 the user is more likely to quickly click the "Request password" button instead of the little arrow to submit the current one.

You're right, but I have no idea how to improve that. I hope for a good suggestion from @jancborchardt 

> After step 7, an extra click is needed to go back.

You're right, I'll see how I can improve that.

**Comment by PVince81** [comment]
I just got a few suggestions from him an hour ago, posting here:
- [ ] "Request password" button should be just a text link saying "Forgot password?" (just like on log in page) cause otherwise the button seems like the thing to click
- [ ] Same for "Back" – needs to be a text link because otherwise it’s clicked by mistake
- [ ] Would be nice to have a link in the "forgot password" email to directly be able to click
- [ ] Communicate expiration date:
	- [ ] Line below "Password protect" input field saying "Password will expire after 1 hour"
	- [ ] Also in the email, adding to the sentence "It is protected with the following password:" – change to: "It is protected with the following password, which will expire after 1 hour:"

**Comment by PVince81** [comment]
I already had a quick look and if you change the "button" to a "a href='#'" with no attribute it gets the same style like on the login page, so already looking good.

@StCyr great if you can look into those

**Comment by StCyr** [comment]
> I just got a few suggestions from him an hour ago, posting here:
> 
>     * [ ]  "Request password" button should be just a text link saying "Forgot password?" (just like on log in page) cause otherwise the button seems like the thing to click

Done 

> 
>     * [ ]  Same for "Back" – needs to be a text link because otherwise it’s clicked by mistake
> 

Done

>     * [ ]  Would be nice to have a link in the "forgot password" email to directly be able to click

I believe that we don't want to communicate the share's link in the same email as the share's password. Could you please confirm?

> 
>     * [ ]  Communicate expiration date:
>       
>       * [ ]  Add line below "Password protect" input field in the share dropdown in files sidebar, saying "Password will expire after 1 hour"

Done

>       * [ ]  Also in the email, adding to the sentence "It is protected with the following password:" – change to: "It is protected with the following password, which will expire after 1 hour:"

Changed to: 

```
It is protected with the following password:

ztI0KjBmKNrLwBN3yI+D

This password will expire at Thu, 14 Apr 2022 20:28:27 +0000
```

**Comment by PVince81** [comment]
@StCyr thanks a lot for the PR!

> I believe that we don't want to communicate the share's link in the same email as the share's password. Could you please confirm?

yeah, you're right, we should keep them separate

**Comment by StCyr** [comment]
Hi @blizzz ,

Isn't it still possible to push these changes in NC24?

**Comment by PVince81** [comment]
@StCyr we're already past feature freeze and in RC2, so we only merge critical changes to avoid destabilizing the build

these changes are only cosmetic, so we can ship them with 24.0.1 later in May

**Comment by PVince81** [comment]
> When the person types in the password and it has expired, the warning message should not say that it is simply wrong, but rather that it has expired and that a new one should be requested.

Usually for security it is sometimes best to not tell the user whether the password is invalid or has expired.

Maybe the message could be changed to something like "the password is wrong or has expired, please request a new one"

@StCyr @jancborchardt thoughts ?

**Comment by Jerome-Herbinet** [comment]
This would be a good compromise.

**Comment by Jerome-Herbinet** [comment]
> > When the person types in the password and it has expired, the warning message should not say that it is simply wrong, but rather that it has expired and that a new one should be requested.
> 
> Usually for security it is sometimes best to not tell the user whether the password is invalid or has expired.
> 
> Maybe the message could be changed to something like "the password is wrong or has expired, please request a new one"
> 
> @StCyr @jancborchardt thoughts ?

This would be a good compromise.
